### PR TITLE
Mes-8824-testSlotTruncateName

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed a bug where the test slot names would not truncate correctly
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image

### CODE REVIEW

**Review by gittins7 — APPROVED** [review]

File: src/components/test-slot/candidate-link/candidate-link.ts [hunk 1]
```diff
@@ -30,15 +37,85 @@ export class CandidateLinkComponent {
...
+
+  truncateValue() {
+    let finalSum = 0;
+
+    switch (this.app.getTextZoomClass()) {
+      case 'text-zoom-regular':
+        finalSum += 35;
+        if (!this.isPortrait) {
+          finalSum += 13;
+        }
```
Comment (matthew2564) [inline src/components/test-slot/candidate-link/candidate-link.ts]: Can we make these a little nicer on the eye

`finalSum += (this.isPortrait ? 35 : 13);`

File: src/components/test-slot/candidate-link/candidate-link.ts [hunk 2]
```diff
@@ -30,15 +37,85 @@ export class CandidateLinkComponent {
...
+        break;
+      default:
+        finalSum += 100;
+        if (!this.isPortrait) {
+          finalSum += 30;
+        }
+        break;
+    }
+
+    finalSum -= (this.slot.booking.application.testCategory.length);
```
Comment (matthew2564) [inline src/components/test-slot/candidate-link/candidate-link.ts]: We need to add some guards to accessing this data, for non test slots we wont have an `application` object

File: src/components/test-slot/candidate-link/candidate-link.ts (hunk 1 above)
Comment (RLCorp) [reply src/components/test-slot/candidate-link/candidate-link.ts]: solution refactored

File: src/components/test-slot/candidate-link/candidate-link.ts (hunk 2 above)
Comment (RLCorp) [reply src/components/test-slot/candidate-link/candidate-link.ts]: solution refactored

**Review by matthew2564 — APPROVED** [review]

File: src/app/pages/fake-journal/fake-journal.page.ts [hunk 3]
```diff
@@ -24,14 +25,16 @@ export class FakeJournalPage extends BasePageComponent {
...
     super(platform, authenticationProvider, router);
 
     this.dateToDisplay = moment().format('dddd D MMMM YYYY');
   }
 
-  ionViewDidEnter(): void {
+  async ionViewWillEnter() {
     this.store$.dispatch(FakeJournalDidEnter());
+    await this.orientationMonitorProvider.monitorOrientation();
   }
```
Comment (matthew2564) [inline src/app/pages/fake-journal/fake-journal.page.ts]: Can we undo this, we can have a did and will enter

File: src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.ts [hunk 4]
```diff
@@ -67,7 +68,10 @@ export class CandidateSearchCardComponent implements OnChanges {
   selectedCandidate: CandidateData;
   enableShowBookingButton: boolean = false;
 
-  constructor(private store$: Store<StoreModel>) {}
+  constructor(
+    private store$: Store<StoreModel>,
+    public orientationMonitorProvider: OrientationMonitorProvider,
+  ) {}
```
Comment (matthew2564) [inline src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.ts:74]: Is this needed?

File: src/app/pages/test-centre-journal/components/view-journals-card/view-journals-card.ts [hunk 5]
```diff
@@ -81,6 +82,7 @@ export class ViewJournalsCardComponent implements OnChanges {
   slotItems$: Observable<SlotItem[]>;
 
   constructor(
+    public orientationMonitorProvider: OrientationMonitorProvider,
```
Comment (matthew2564) [inline src/app/pages/test-centre-journal/components/view-journals-card/view-journals-card.ts:85]: Is this used?

File: src/app/providers/orientation-monitor/orientation-monitor.provider.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+export class OrientationMonitorProvider {
+
+  isPortraitMode$: BehaviorSubject<boolean> = new BehaviorSubject(false);
+
+  constructor(
+    private ref: ApplicationRef,
+  ) {
+  }
+
+  async ionViewWillLeave() {
```
Comment (matthew2564) [inline src/app/providers/orientation-monitor/orientation-monitor.provider.ts]: Just checking this actually fires? Wasnt aware life cycle hooks applied to providers

File: src/app/pages/test-centre-journal/components/view-journals-card/view-journals-card.ts (hunk 5 above)
Comment (RLCorp) [reply src/app/pages/test-centre-journal/components/view-journals-card/view-journals-card.ts:85]: it's used to call isPortrait in html

File: src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.ts (hunk 4 above)
Comment (RLCorp) [reply src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.ts:74]: it's used to call isPortrait in html

File: src/app/providers/orientation-monitor/orientation-monitor.provider.ts (hunk 6 above)
Comment (RLCorp) [reply src/app/providers/orientation-monitor/orientation-monitor.provider.ts]: moved in newest commit

File: src/app/pages/fake-journal/fake-journal.page.ts (hunk 3 above)
Comment (RLCorp) [reply src/app/pages/fake-journal/fake-journal.page.ts]: changed in newest commit

File: src/app/providers/orientation-monitor/__tests__/orientation-monitor.provider.spec.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    provider = TestBed.inject(OrientationMonitorProvider);
+  });
+
+  describe('tearDownListener', () => {
+    it('should call removeAllListeners', () => {
+      spyOn(ScreenOrientation, 'removeAllListeners');
+      provider.tearDownListener();
+      expect(ScreenOrientation.removeAllListeners());
```
Comment (matthew2564) [inline src/app/providers/orientation-monitor/__tests__/orientation-monitor.provider.spec.ts]: There is no expectation here

Comment (RLCorp) [reply src/app/providers/orientation-monitor/__tests__/orientation-monitor.provider.spec.ts]: fixed in new commit

**Pull request merged**